### PR TITLE
Add Topology Button To Individual Region Pages #493

### LINKED ISSUE #493: Add Topology Button To Individual Region Pages
### NetBox version

v3.7.8

### Topology Views version

v3.9.1

### Feature type

New functionality

### Proposed functionality

Just like the plug-in adds a Topology Button to each Site, which then draws/shows the topology for the site, this request is the do the same thing for each Region

### Use case

The plug-in already has the ability to show regions when using the plugin, but it would be the same use case as showing the site topology button for each individual sites, but for regions. It would increase functionality of the tool and use cases.

### External dependencies

Not sure, but not really since it’s an expansion of current functionality.

### PR DESCRIPTION
**Pull request by mr1716**
Closes #493 

This will add a Topology Button to the Region Page

### CODE REVIEW

**Comment by dreng** [comment]
Hi,

thank you for your contribution!

> This will add a Topology Button to the Region Page

I'm afraid it doesn't. You've created a template but you have omitted to create a class for the template content.

File: netbox_topology_views/templates/netbox_topology_views/region_button.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<a role="button" class="btn btn-sm btn-primary" title="Topology" href="#" hx-get="/plugins/netbox_topology_views/topology/?region_id={{ object.id }}&show_cables=on&show_unconnected=on" hx-target="#htmx-modal-content" data-bs-toggle="modal" data-bs-target="#htmx-modal">
+    <i class="mdi mdi-transit-connection-variant" aria-hidden="true"></i>
+    Topology
+</a>
```
Comment (dreng) [inline netbox_topology_views/templates/netbox_topology_views/region_button.html:4]: Always put a newline to the end of a file.

**Review by dreng — CHANGES_REQUESTED** [review]

**Comment by mr1716** [comment]
@dreng thanks. Change made and pushed . Please review

**Comment by dreng** [comment]
> @dreng thanks. Change made and pushed . Please review

You added a newline, which is good. You did not notice my first comment, right? You need to add a class for the template content. Otherwise the button will not be shown.

**Comment by mr1716** [comment]
@dreng looks like I want to make some improvements to it and the product such as different colors for the group boundaries. So I’ll close the request.

**Pull request closed without merging**